packages/turf-boolean/touch.js

```diff
@@ -0,0 +1,98 @@
+var checkIfCoordArraysMatch = internalHelpers.checkIfCoordArraysMatch;
+var isPointOnLineSegment = internalHelpers.isPointOnLineSegment;
+
+module.exports = function (feature1, feature2) {
+    if (feature1.geometry.type === 'Point' && feature2.geometry.type === 'LineString') {
+        return isPointAtEndOfLine(feature1, feature2);
+    }
+    if (feature1.geometry.type === 'MultiPoint' && feature2.geometry.type === 'LineString') {
+        return isMultiPointAtEndOfLine(feature1, feature2);
+    }
+    if (feature1.geometry.type === 'Point' && feature2.geometry.type === 'Polygon') {
+        return isPointOnPolygon(feature1, feature2);
+    }
+    if (feature1.geometry.type === 'MultiPoint' && feature2.geometry.type === 'Polygon') {
+        return isMultiPointOnPolygon(feature1, feature2);
+    }
+    // if (feature1.geometry.type === 'LineString' && feature2.geometry.type === 'LineString') {
+    //     return isLineStringOnLineString(feature1, feature2);
+    // }
+    // if (feature1.geometry.type === 'LineString' && feature2.geometry.type === 'Polygon') {
+    //     return isLineStringOnPolygon(feature1, feature2);
+    // }
+};
+
+function isPointAtEndOfLine(Point, LineString) {
+    if (checkIfCoordArraysMatch(Point.geometry.coordinates, LineString.geometry.coordinates[0])) {
+        return true;
+    }
+    if (checkIfCoordArraysMatch(Point.geometry.coordinates, LineString.geometry.coordinates[LineString.geometry.coordinates.length - 1])) {
+        return true;
+    }
+    return false;
+}
+
+function isMultiPointAtEndOfLine(MultiPoint, LineString) {
+    var i;
+    var onStartOrEnd = 0;
+    var pointIsSomewhereOnLine = 0;
+    var lastLinestringCoordIndex = LineString.geometry.coordinates.length - 1;
+    for (i = 0; i < MultiPoint.geometry.coordinates.length; i++) {
+        if (checkIfCoordArraysMatch(MultiPoint.geometry.coordinates[i], LineString.geometry.coordinates[0])) {
+            onStartOrEnd++;
+            continue;
+        }
+        if (checkIfCoordArraysMatch(MultiPoint.geometry.coordinates[i], lastLinestringCoordIndex)) {
+            onStartOrEnd++;
+            continue;
+        }
+        for (var lineI = 0; lineI < LineString.geometry.coordinates.length - 1; lineI++) {
+            if (isPointOnLineSegment(LineString.geometry.coordinates[lineI], LineString.geometry.coordinates[lineI + 1], MultiPoint.geometry.coordinates[i])) {
+                pointIsSomewhereOnLine++;
+            }
+        }
+    }
+    return onStartOrEnd === 1 && pointIsSomewhereOnLine === 0;
+}
+
+function isPointOnPolygon(Point, Polygon) {
+    var output = false;
+    var i;
+    for (i = 0; i < Polygon.geometry.coordinates[0].length - 1; i++) {
+        if (isPointOnLineSegment(Polygon.geometry.coordinates[0][i], Polygon.geometry.coordinates[0][i + 1], Point.geometry.coordinates)) {
+            output = true;
+            break;
+        }
+    }
+    return output;
+}
+
+function isMultiPointOnPolygon(MultiPoint, Polygon) {
```

Comment:
I think you also need a check that none of the points are in the interior of the polygon.